fe/src/test/java/org/apache/doris/load/loadv2/HllTest.java

```diff
@@ -0,0 +1,268 @@
+        Assert.assertTrue(explicitHll.estimateCardinality() == HLL_EXPLICLIT_INT64_NUM);
+
+        ByteArrayOutputStream explicitOutputStream = new ByteArrayOutputStream();
+        DataOutput explicitOutput = new DataOutputStream(explicitOutputStream);
+        explicitHll.serialize(explicitOutput);
+        DataInputStream explicitInputStream = new DataInputStream(new ByteArrayInputStream(explicitOutputStream.toByteArray()));
+        Hll deserializedExplicitHll = new Hll();
+        deserializedExplicitHll.deserialize(explicitInputStream);
+        Assert.assertTrue(deserializedExplicitHll.getType() == HLL_DATA_EXPLICIT);
+
+        // test sparse
+        Hll sparseHll = new Hll();
+        for (int i = 0; i < HLL_SPARSE_THRESHOLD; i++) {
+            sparseHll.updateWithHash(i);
+        }
+        Assert.assertTrue(sparseHll.getType() == HLL_DATA_FULL);
+        // 2% error rate
+        Assert.assertTrue(sparseHll.estimateCardinality() > HLL_SPARSE_THRESHOLD * (1 - 0.02) &&
+                sparseHll.estimateCardinality() < HLL_SPARSE_THRESHOLD * (1 + 0.02));
+
+        ByteArrayOutputStream sparseOutputStream = new ByteArrayOutputStream();
+        DataOutput sparseOutput = new DataOutputStream(sparseOutputStream);
+        sparseHll.serialize(sparseOutput);
+        DataInputStream sparseInputStream = new DataInputStream(new ByteArrayInputStream(sparseOutputStream.toByteArray()));
+        Hll deserializedSparseHll = new Hll();
+        deserializedSparseHll.deserialize(sparseInputStream);
+        Assert.assertTrue(deserializedSparseHll.getType() == HLL_DATA_SPARSE);
+        Assert.assertTrue(sparseHll.estimateCardinality() == deserializedSparseHll.estimateCardinality());
+
+
+        // test full
+        Hll fullHll = new Hll();
+        for (int i = 1; i <= Short.MAX_VALUE; i++) {
+            fullHll.updateWithHash(i);
+        }
+        Assert.assertTrue(fullHll.getType() == HLL_DATA_FULL);
+        // the result 32748 is consistent with C++ 's implementation
+        Assert.assertTrue(fullHll.estimateCardinality() == 32748);
+        Assert.assertTrue(fullHll.estimateCardinality() > Short.MAX_VALUE * (1 - 0.02) &&
+                fullHll.estimateCardinality() < Short.MAX_VALUE * (1 + 0.02));
+
+        ByteArrayOutputStream fullHllOutputStream = new ByteArrayOutputStream();
+        DataOutput fullHllOutput = new DataOutputStream(fullHllOutputStream);
+        fullHll.serialize(fullHllOutput);
+        DataInputStream fullHllInputStream = new DataInputStream(new ByteArrayInputStream(fullHllOutputStream.toByteArray()));
+        Hll deserializedFullHll = new Hll();
+        deserializedFullHll.deserialize(fullHllInputStream);
+        Assert.assertTrue(deserializedFullHll.getType() == HLL_DATA_FULL);
+        Assert.assertTrue(deserializedFullHll.estimateCardinality() == fullHll.estimateCardinality());
+
+    }
+
+    // keep logic same with C++ version
+    // add additional compare logic with C++ version's estimateValue
```

Comment:
better to add comment in `be/test/olap/hll_test.cpp` to notice that the following test case should be consistent with java version in `fe/src/test/java/org/apache/doris/load/loadv2/HllTest.java`

Reply:
👌